src/foundry/foundry.js/polygonVertex.d.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * An internal data structure for polygon vertices
+ * @internal
+ */
+declare class PolygonVertex {
+  constructor(x: number, y: number, { distance, index }?: PolygonVertexOptions);
+
+  x: number;
+
+  y: number;
+
+  key: number;
+
+  protected _distance: unknown;
+
+  protected _d2: unknown;
+
+  protected _index: unknown;
```

Comment:
These should typed properly (or if it's not possible yet, FIXMEs should be added).